---
layout: post
title:  "Differentially expressed gene in cell clusters by k-means"
author: April Yan
jhed: yyan67
categories: [ HW4 ]
image: homework/hw4/hw4_yyan67.png
featured: false
---

## Write a description to convince me that your cluster interpretation is correct. Your description may reference papers and content that allowed you to interpret your cell cluster as a particular cell-type. 

The top left panel demonstrates how k-means clusters are distributed spatially in physical space (aligned_y vs. aligned_x). I choose k=4 randomly. The top middle panel only selected the cluster of interest. The top right panel is the PC plot of the selected cell cluster. The principal components are calculated based on all the expression data of each gene. It seems that there could be two sub-clusters in this cluster, one of which tend to have smaller values for PC1 and the other one tend to have smaller values for PC2. 

We have also plotted the p-values for the top 50 genes with most significant differentially expression level compared with cells in other clusters (see bottom left panel where they were all very close zero). I just randomly picked one called ACTA2 (usually expressed in smooth muscle cells). I plotted all cells in t-SNE embedding space generated by normalized gene expression of all genes. I labeled the cells with normalized ACTA2 expression. As we can see from the bottom middle panel, there could be certain clusterson the top right corner with lighter blue (i.e., high expression of ACTA2) that are distinct from others with darker colors. However, there were more than just one cluster in t-SNE so there could be subtypes in this cell cluster. The expression of ACTA2 were also demonstrated in physically space (see bottom right panel). The spatial positions of cells with highly expressed ACTA2 overlap in general with the cells in the cluster of interest. 

As a result, based on all the visualizations, I conclude that this one type of cell could be smooth muscle cells with a couple of subtypes. 

## Please share the code you used to reproduce this data visualization.
```{r}
data <- read.csv('pikachu.csv.gz', row.names = 1)
library(ggplot2)
library(patchwork)

pos <- data[, 4:5]
total_count <- rowSums(data[ ,6:ncol(data)], na.rm=TRUE)
gexp <- data[, 6:ncol(data)]
km <- kmeans(gexp, centers=4)
df <- data.frame(data, kmeans=as.factor(km$cluster))
head(df)
p0 = ggplot(df) + geom_point(aes(x = aligned_x, y=aligned_y, col=kmeans), 
             size=0.1, alpha=0.5) + theme_minimal()

# cluster of interest: 3
df1 = df[df$kmean == 3,]
p1 = ggplot(df1) + geom_point(aes(x = aligned_x, y=aligned_y, col=kmeans), 
             size=0.1, alpha=0.5) + theme_minimal()

pca = data.frame(prcomp(gexp)$x[,1:2])
p2 = ggplot(pca) + geom_point(aes(x = PC1, y=PC2), 
                              size=0.1) + theme_minimal()

# df2 = df[df$kmean != 3,]
gexpnorm <- log10(gexp/rowSums(gexp) * mean(rowSums(gexp))+1)
# com <- as.factor(kmeans(gexpnorm, centers=4)$cluster)
results <- sapply(colnames(gexpnorm), function(g) {
  wilcox.test(gexpnorm[km$cluster == 3, g],
              gexpnorm[km$cluster != 3, g],
              alternative = "greater")$p.val})
pval = data.frame(pval = sort(results, decreasing=FALSE)[1:50])
pval$gene = row.names(pval)
par(mar=c(1,1,1,1))
p3 = ggplot(pval) + geom_point(aes(x=gene, y=pval)) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size=5))

pcs <- prcomp(gexpnorm)
emb <- Rtsne(pcs$x[,1:20])$Y
p4 = ggplot(data.frame(emb, gene = gexpnorm[,'ACTA2'])) + 
  geom_point(aes(x = X1, y = X2, col=gene), size=0.01, alpha=0.5) + 
  theme_bw()

p5 = ggplot(data.frame(data[, 4:5], gene = gexpnorm[,'ACTA2'])) + 
  geom_point(aes(x=aligned_x, y=aligned_y, col=gene), size=0.01, alpha=0.5) + 
  theme_bw()

p0 + p1 + p2 + p3 + p4 + p5
``` 

## external resources
https://en.wikipedia.org/wiki/ACTA2
