---
layout: post
title:  "Analyzing for Breast Glandular Cells"
author: Jonathan Wang
jhed: jwang428
categories: [ HW4 ]
image: homework/hw4/hw4_jwang428.png
featured: false
---

## Cell Type Believed to Have Identified:
I believe that my analysis of Cluster 9 within the Eevee Dataset reveals that the 
cluster is comprised of Breast Glandular Cells, more specifically of the epithelial
cell type. 

## Reasoning: 
The top 5 most significantly upregulated genes in Cluster 9 were as follows: CGN,
EPS8L1, FAM83B, KRT80, and FAM3B. If we query each of these genes on the human protein
atlas (https://www.proteinatlas.org/), we can see that a common tissue cell type across
all 5 genes is the Breast Glandular Cell Type. You can find the associated links to each
of the 5 queries below [1-5]. Various publications have also linked the 5 genes to being 
associated with breast cancer tissue samples. For CGN, it has been observed to be 
downregulated in breast cancer cells [6].  FOR EPS8L1, it has been observed to be
upregulated in breast cancer cells [7]. For FAM83B and FAM3B, researchers have identified 
both as key biomarkers of prognostic signatures for triple-negative breast cancer [8, 10]. 
For KRT80, it has been noted to be upregulated in endocrine resistant ERÎ± breast cancer [9]. 

I also mention that the genes are also linked to the more broad cell type group of epithelial
cells due to searching each of the genes on NCBI's database (https://www.ncbi.nlm.nih.gov), 
where the queries have all pointed to the genes being most expressed in epithelial cells 
throughout the body. What made me confident that these epithelial cells are Breast Glandular Cells
is the (1) protein atlas database and (2) the numerous research papers also linking the 
upregulated genes to breast cancer tissue. 

## Links / References:
[1] CGN on Protein Atlas: https://www.proteinatlas.org/ENSG00000143375-CGN/tissue+cell+type/  
[2] EPS8L1 on Protein Atlas: https://www.proteinatlas.org/ENSG00000131037-EPS8L1/tissue+cell+type   
[3] FAM83B on Protein Atlas: https://www.proteinatlas.org/ENSG00000168143-FAM83B/tissue+cell+type      
[4] FAM83B on Protein Atlas: https://www.proteinatlas.org/ENSG00000167767-KRT80/tissue+cell+type   
[5] FAM3B on Protein Atlas: https://www.proteinatlas.org/ENSG00000183844-FAM3B/tissue+cell+type   
[6] 10.3389/fmolb.2022.758974  
[7] 10.3892/or.2021.7927   
[8] 10.1186/s12885-021-08318-1   
[9] 10.1038/s41467-019-09676-y   
[10] 10.3389/fphar.2023.1191910   

```{r}
## Packages for graphing
library(ggplot2)
library(dplyr)
library(gridExtra)
library(ggrepel)

## Packages for determining optimal cluster number
library(factoextra)
library(NbClust) 
library(cluster)

## Load the data
data <- read.csv('eevee.csv.gz', row.names = 1)

## Isolate columns of interest
pos <- data[,2:3]
gexp <- data[, 4:ncol(data)]

## Isolate cells that have gene Expression
good_cells <- rownames(gexp)[rowSums(gexp) > 0]
pos <- pos[good_cells,]
gexp <- gexp[good_cells,]

## Normalize the data
tot_gexp <- rowSums(gexp)
norm_gexp <- gexp/tot_gexp * median(tot_gexp)
log_norm_gexp <- log10(norm_gexp + 1)

## Run PCA on the data
pcs <- prcomp(log_norm_gexp)

## Determine optimal number of PCs to run for kMeans
plot(1:50, pcs$sdev[1:50], type = 'l')
plot(1:20, pcs$sdev[1:20], type = 'l')
plot(1:10, pcs$sdev[1:10], type = 'l')

## Appears that having 9 PCs is enough to account for most
## of the variance in the data.
optimal_pc_num <- 9 

## Determine the optimal number of k's (centroids) for kMeans

## Gap Statistic
set.seed(123)
gap_stat <- clusGap(log_norm_gexp, FUNcluster = kmeans, K.max = 10, B = 50)
fviz_gap_stat(gap_stat)

## Identifying optimal number of k's
optimal_k <- 10 
  
## Set seed for reproducibility
set.seed(1)

## Run kMeans
cluster_pcs <- kmeans(pcs$x[,1:optimal_pc_num], centers = optimal_k)
df_cluster <- data.frame(pos, cell_type = as.factor(cluster_pcs$cluster), gexp, pcs$x[,1:10])

## Visualizing cells in the 2D space
clusters_2D <- ggplot(data = df_cluster, aes(x = aligned_x, y = aligned_y, col = cell_type)) + 
  geom_point(size = 1) + 
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab("Aligned X Position") + ylab("Aligned Y Position") +
  guides(color = guide_legend(override.aes = list(size = 2))) +
  labs(title = 'Clusters in Physical 2D Space', col = "Cell Type")

clusters_2D

## Visualizing cells in the PCA space
clusters_PCA <- ggplot(data = df_cluster, aes(x = PC1, y = PC2, col = cell_type)) + 
  geom_point(size = 1) + 
  theme_bw() + 
  theme(plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(size = 2))) +
  labs(title = 'Clusters in PCA Space', col = "Cell Type")

clusters_PCA

## Picking a cluster (cluster 9)

## Visualizing Cluster 9 in Physical 2D Space
cluster9_2D <- ggplot(data = df_cluster, aes(x = aligned_x, y = aligned_y, col = cell_type == 9)) + 
  geom_point(size = 1) + 
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(size = 2))) +
  labs(title = 'Cluster 9 in Physical 2D Space') +
  xlab("Aligned X Position") + ylab("Aligned Y Position") +
  scale_color_manual(name = 'Cell Type',
                     values = c('lightgrey', '#D772EC'),
                     labels = c('Other Cell Type', 'Cell Type 9'))

cluster9_2D

## Visualizing Cluster 9 in the PCA Space
cluster9_PCA <- ggplot(data = df_cluster, aes(x = PC1, y = PC2, col = cell_type == 9)) + 
  geom_point(size = 1) + 
  theme_bw() + 
  theme(plot.title = element_text(hjust = 0.5)) +
  guides(color = guide_legend(override.aes = list(size = 2))) +
  labs(title = 'Cluster 9 in PCA Space') +
  scale_color_manual(name = 'Cluster',
                     values = c('lightgrey', '#D772EC'),
                     labels = c('Other Cell Type', 'Cell Type 9'))

cluster9_PCA

## Distinguishing between cells that are & are not within Cluster 9
cluster9 <- df_cluster[df_cluster$cell_type == 9, ]
not_cluster9 <- df_cluster[df_cluster$cell_type != 9, ]

## Identifying differentially expressed genes w/in Cluster 9 w/ Wilcox Rank Test
interest_genes <- sapply(colnames(gexp), function(g){ 
  wilcox.test(gexp[row.names(cluster9), g], 
              gexp[row.names(not_cluster9), g])$p.value  
})

## Calculating log(fold_change) for Cluster 9 --> to generate volcano plot
log_fc <- sapply(colnames(gexp), function(g) {
  log2( mean(gexp[row.names(cluster9), g]) 
        / mean(gexp[row.names(not_cluster9), g])) ## do ratio (log_2(A/B))
})

volcano <- data.frame(log_fc, log_p_val = -log10(interest_genes))
genes_upregulated <- sapply(colnames(gexp), function(g){
  wilcox.test(gexp[row.names(cluster9), g], 
              gexp[row.names(not_cluster9), g], 
              alternative = 'greater')$p.value
})

top_20_genes <- sort(genes_upregulated, decreasing = FALSE)[1:20]
top_20_genes

## For coloring in the points for volcano plot
volcano$diff_express <- "NOT SIGNIFICANT"
volcano$diff_express[volcano$log_fc > 0.6 & volcano$log_p_val > -log10(0.05)] <- "UP REGULATED"
volcano$diff_express[volcano$log_fc < -0.6 & volcano$log_p_val > -log10(0.05)] <- "DOWN REGULATED"

## Creating volcano plot to visualize upregulated genes in Cluster 9! 
volcano_plot <- ggplot(volcano) + 
  geom_point(aes(x = log_fc, y = log_p_val, col = diff_express)) +
  geom_vline(xintercept = c(-0.6, 0.6), linetype = 'dashed') + 
  geom_hline(yintercept=-log10(0.05), linetype = 'dashed') +
  geom_label_repel(data= volcano[names(top_20_genes),],
                   aes(x = log_fc, y = log_p_val, label = names(top_20_genes)), 
                   max.overlaps = 40) +
  theme_bw() + 
  labs(title = 'Differentially Expressed Genes: Cluster 9',
       x = 'Log(Fold Change)', y = 'Log(P-Value)', col = "Gene Expression") + 
  theme(plot.title = element_text(hjust = 0.5))

volcano_plot

## Plot the top 2 most significant genes (CGN, EPS8L1)

## Plotting in Physical 2D Space
gene_2d <- ggplot(data = data, aes(x = aligned_x, y = aligned_y, col = CGN, size = EPS8L1)) +
  geom_point() + 
  theme_bw() + 
  scale_color_gradient(low = 'lightgrey', high = 'blue') + 
  scale_size(range = c(0.5, 5)) +
  labs(x = "Aligned X Position", y = "Algned Y Position", size = "EPS8L1", color = "CGN", title = "CGN & EPS8L1 Expression in Physical 2D Space") + 
  theme(plot.title = element_text(hjust = 0.5)) 

gene_2d

## Plotting in PCA Space
gene_pca <- ggplot(data = df_cluster, aes(x = PC1, y = PC2, col = CGN, size = EPS8L1)) +
  geom_point() + 
  theme_bw() + 
  scale_color_gradient(low = 'lightgrey', high = 'blue') + 
  scale_size(range = c(0.5, 5)) +
  labs(size = "EPS8L1", color = "CGN", title = "CGN & EPS8L1 Expression in PCA Space") + 
  theme(plot.title = element_text(hjust = 0.5)) 

gene_pca

## Combining all the plots together
lay <- rbind(c(1,2),
             c(3,4),
             c(5,6),
             c(7,7))

grid.arrange(clusters_2D, clusters_PCA,
             cluster9_2D, cluster9_PCA,
             gene_2d, gene_pca,
             volcano_plot,
             layout_matrix = lay,
             top = "Analyzing the Cell Identity of Cluster 9: Breast Glandular Cells")

## References:
## (1) To determine the optimal number of clusters
## https://medium.com/@chyun55555/how-to-find-the-optimal-number-of-clusters-with-r-dbf84988388b 
## (2) Graphing volcano plots
## https://biocorecrg.github.io/CRG_RIntroduction/volcano-plots.html

```