---
layout: post
title:  "Visualizing Changes in Clustering from t-SNE Transformation Across Key Principal Component Pairs: Exploring Extremes and Intermediates"
author: Krishna Mukunda
jhed: kmukund1
categories: [ HW3 ]
image: homework/hw2/hw2_kmukund1.png
featured: false
---

### What data types are you visualizing?
Gene Expression: Quantitative.
X,Y positions: Quantitative.
Principle Components: Ordinal.

I am visualizing the changes in percieved clustering from non-linear dimensional reduction by t-SNE on Principle Component analysis of gene expression of the most variable gene in the eevee data set. The gene expression data is quantitative and discrete with each value representing the level of gene expression in each cell. The X,Y positions are quantitative data as well with their values corresponding to the position of the center of each cell. The PCs are ordinal as their values (1,2,3,..) indicate progressively lower varianc. 

### What data encodings (geometric primitives and visual channels) are you using to visualize these data types?
Geometric Principles: Points.
Visual Channels: Position, Size, Color (Hue).

THe visualization contains points that represent each cell in the data set in plots b-h, specifically, their x,y centroid positions. Plot "a" uses points as well which each pc with their position along the y axis indicating their   variance. Positions in plot "b-h" reperesent each cells spatial relation following linear and non-linear transforms. Color through hue is used to indicate gene expression in plots b-h with red being high expression and green being low expression.

### What about the data are you trying to make salient through this data visualization? 
My visualization aims to make more salient the effects of higher PC values on the percieved clustering of the cells. From the scree plot, we see that the higher principle components have lower variance. The t-SNE were performed on the first, middle, and last PCs. Non-linear dimensionality reduction on PCs 1 and 2 showed high levels of clustering of cells with low gene expression while the middle and last PCs showed nearly indistinguishable areas of high and low expression of the most variable gene. 


### What Gestalt principles or knowledge about perceptiveness of visual encodings are you using to accomplish this?
Similarity: Cells that have similar color are seen as having similar gene expression and may indicate clusters.

Enclosure: Each plot has its own axis which encloses them from the other plots to indicate that they are separate. 

Proximity: Plots that analyzed the same principle componets were place in the same column so that they may be percieved as part of the same group.

```{r}
library(Rtsne)
library(ggplot2)
library(patchwork)

data <- read.csv("/Users/knm/github/genomic-data-visualization-2024/data/eevee.csv.gz", row.names = 1)
gexp <- data[, 4:ncol(data)]
topgene <- names(sort(apply(gexp, 2, var), decreasing = TRUE))
gexpfilter <- gexp[,topgene]

pcs <- prcomp(gexpfilter)
dim(pcs$x)

N <- ncol(pcs$x)

emb1 <- Rtsne(pcs$x[,1:2], dims = 2, perplexity = 50)
middle_start <- floor((N - 1) / 2)
middle_end <- middle_start + 1
emb2 <- Rtsne(pcs$x[,middle_start:middle_end], dims = 2, perplexity = 50)

emb3 <- Rtsne(pcs$x[,(N-1):N], dims = 2, perplexity = 50)



df <- data.frame(PC1_1 = pcs$x[,1], PC1_2 = pcs$x[,2], 
                 PC2_1 = pcs$x[,middle_start], PC2_2 = pcs$x[,middle_end],
                 PC3_1 = pcs$x[,N], PC3_2 = pcs$x[,(N-1)], 
                 emb1_1 = emb1$Y[,1], emb1_2 = emb1$Y[,2],
                 emb2_1 = emb2$Y[,1], emb2_2 = emb2$Y[,2],
                 emb3_1 = emb3$Y[,1], emb3_2 = emb3$Y[,2],
                 gene = rowSums(gexpfilter), x = data$aligned_x, y = data$aligned_y)


p1 <- ggplot(df) +
  geom_point(aes(x = emb1_1, y = emb1_2, color = log10(gene)), alpha = 0.7) +
  scale_color_gradient(low = "green", high = "red") +
  theme_classic() +
  labs(title = "t-SNE on PCs 1,2", color = "Log10 Gene Exp.", 
       x = "Component 1", y = "Component 2")

p2 <-ggplot(df) + 
  geom_point(aes(x = emb2_1, y = emb2_2, color = log10(gene))) +
  scale_color_gradient(low = "green", high = "red") +
  theme_classic() +
  labs(title = "t-SNE on PCs 354,355", color = "Log10 Gene Exp.", 
       x = "Component 1", y = "Component 2")

p3 <-ggplot(df) + 
  geom_point(aes(x = emb3_1, y = emb3_2, color = log10(gene))) +
  scale_color_gradient(low = "green", high = "red") +
  theme_classic() +
  labs(title = "t-SNE on PCs 708,709", color = "Log10 Gene Exp.", 
       x = "Component 1", y = "Component 2")

p4 <- ggplot(df) + 
  geom_point(aes(x = PC1_1, y = PC1_2, color = log10(gene))) +
  scale_color_gradient(low = "green", high = "red") +
  theme_classic() +
  labs(title = "PCs 1,2", color = "Log10 Gene Exp.",
       x = "PC 1", y = "PC 2")

p5 <- ggplot(df) + 
  geom_point(aes(x = PC2_1, y = PC2_2, color = log10(gene))) +
  scale_color_gradient(low = "green", high = "red") +
  theme_classic() +
  labs(title = "PCs 354,355", color = "Log10 Gene Exp.",
       x = "PC 354", y = "PC 355")

p6 <- ggplot(df) + 
  geom_point(aes(x = PC3_1, y = PC3_2, color = log10(gene))) +
  scale_color_gradient(low = "green", high = "red") +
  theme_classic() +
  labs(title = "PCs 708,709", color = "Log10 Gene Exp.", 
       x = "PC 708", y = "PC 709")


p7 <- ggplot(df) + 
  geom_point(aes(x = x, y = y, color = log10(gene))) +
  scale_color_gradient(low = "green", high = "red") +
  theme_classic() +
  labs(title = "Cell Positions", color = "Log10 Gene Exp.", 
       x = "X", y= "Y")

variance_explained <- pcs$sdev^2
total_variance <- sum(variance_explained)
proportion_variance_explained <- variance_explained / total_variance
pc_data <- data.frame(PC_Number = 1:length(proportion_variance_explained), 
                      ProportionVariance = proportion_variance_explained)
p8 <- ggplot(pc_data, aes(x = PC_Number, y = ProportionVariance)) +
  geom_line() + 
  geom_point() +
  labs(title = "Scree Plot", x = "Principal Component", y = "Proportion of Variance Explained")

combined_plot <- p8 + p1 + p2 + p3 + p7 + p4 + p5 + p6 + plot_annotation(tag_levels = 'a') + plot_layout(ncol = 4)
combined_plot

ggsave("/Users/knm/Library/CloudStorage/OneDrive-JohnsHopkins/Johns Hopkins [2021-2025]/Junior [2023-2024]/Genomics Data Visualization/hw3_kmukund1.png", combined_plot, width = 20, height = 8)
```

