---
layout: post
title: "Spatially Resolved Gene Expression Analysis using KMeans clustering"
author: Nidhi Soley
jhed: nsoley1
categories: [ HW4 ]
image: homework/hw4/hw1_nsoley1.png
featured: false
---

### Describe your figure briefly so we know what you are depicting (you no longer need to use precise data visualization terms as you have been doing). Write a description to convince me that your cluster interpretation is correct. Your description may reference papers and content that allowed you to interpret your cell cluster as a particular cell-type. You must provide attribution to external resources referenced. Links are fine; formatted references are not required.

Firstly I applied KMeans clustering on the normalized gene expression data. Then, I chose a cluster which was scattered at few spots in spatial position (blue in color), not together like cluster 1 and 2. I was interested to see which gene is highly expressed in this cluster. I performed t-test to see the upregulated genes in this cluster of interest. I found that DSP (pvalue 2.519682e-99) was the top gene that was upregulated. Further, I visualized DSP in the physical space and in PCA space.

1.  **Spatial Distribution of Clusters**: The first panel shows the spatial distribution of spots with multiple cell-types within the tissue. Each spot is represented by a point on the plot, with colors hue indicating the clusters assigned by KMeans clustering. Notably, cluster three (shown in blue) appears scattered across the tissue, contrasting with the more cohesive distributions of clusters one and two.
2.  **Gene Expression in Physical Space**: The second panel displays the expression pattern of the gene DSP across the spatial coordinates. Each point represents a spot, colored based on its expression level of DSP. Notably, cells corresponding to cluster three, which were identified as scattered in spatial position (blue in first plot), exhibit high expression of DSP (yellow).
3.  **Gene Expression in PCA Space**: The third panel illustrates the expression pattern of DSP projected onto a reduced-dimensional space obtained through PCA. Cells within cluster three, which were spatially dispersed, cluster together in this reduced-dimensional space based on their similar expression profiles of DSP.The yellow spots are clustered together on the negative PC1, and PC2 axis.

By integrating spatial information with gene expression analysis, I could infer that cluster three likely represents spot that might have multiple cell type(s) characterized by high expression of the DSP gene. \

The interpretation of cluster three can be done as multiple cell-types exhibiting high DSP expression by aligning with known biology. DSP, or desmoplakin, is a protein involved in cell-cell adhesion, particularly in desmosomes, which are structures found in tissues subjected to mechanical stress, such as the skin and heart. The DSP gene provides instructions for making a protein called desmoplakin. This protein is found primarily in cells of the heart and skin. This interpretation is supported by studies demonstrating DSP expression in tissues such as the epidermis and cardiac muscle [1][2].

[1] Green KJ, Simpson CL. Desmosomes: new perspectives on a classic. J Invest Dermatol. 2007;127(11):2499-2515. <doi:10.1038/sj.jid.5700915>

[2] McGrath JA, Mellerio JE. Ectodermal Dysplasia-Skin Fragility Syndrome. In: Adam MP, Ardinger HH, Pagon RA, et al., editors. GeneReviewsÂ® [Internet]. Seattle (WA): University of Washington, Seattle; 1993-2021. Available from: <https://www.ncbi.nlm.nih.gov/books/NBK33249/>

### You must include the entire code you used to generate the figure so that it can be reproduced.

```{r}
library(viridis)
library(patchwork)
library(ggplot2)
data <- read.csv('/Users/nidhisoley/Desktop/GDataViz/eevee.csv.gz', row.names=1)
pos <- data[, c("aligned_x", "aligned_y")]
gexp <- data[, 4:ncol(data)]
gexpnorm <- log10(gexp/rowSums(gexp) * mean(rowSums(gexp)) + 1)
pcs <- prcomp(gexpnorm)
#K-means
clusters <- as.factor(kmeans(gexpnorm, centers = 3)$cluster)
p1 <- ggplot(data.frame(pos, Cluster = clusters)) + 
  geom_point(aes(x = aligned_x, y = aligned_y, col = Cluster), size = 3) + 
  theme_bw() +
  labs(title = "Clusters in Physical Space")
#differential expression analysis for the cluster of interest
cluster_of_interest <- 3
cluster_1_genes <- colnames(gexpnorm)[clusters == cluster_of_interest]
results <- sapply(cluster_1_genes, function(g) {
  t.test(gexpnorm[clusters == cluster_of_interest, g],
         gexpnorm[clusters != cluster_of_interest, g],
         alternative = 'greater')$p.val
})
significant_genes <- names(head(sort(results[results < 0.05])))
gene_of_interest <- 'DSP'
#physical space
p2 <- ggplot(data.frame(pos, gene = gexpnorm[, gene_of_interest])) +
  geom_point(aes(x = aligned_x, y = aligned_y, col = gene), size = 3) +
  scale_color_viridis(option = 'plasma')+
  theme_bw() +
  labs(title = paste("Expression of", gene_of_interest, "in Physical Space"))
#pca
p3 <- ggplot(data.frame(pcs$x, gene = gexpnorm[, gene_of_interest])) +
  geom_point(aes(x = PC1, y = PC2, col = gene), size = 2) +
  scale_color_viridis(option = 'plasma')+
  theme_bw() +
  labs(title = paste("Expression of", gene_of_interest, "in PCA space"))
p1 +p2+p3+ plot_layout(ncol = 3)
```
