---
layout: post
title: "Normalization of gene expression prior to dimensionality reduction for Spatial Transcriptomic dataset"
author: Nidhi Soley
jhed: nsoley1
categories: [ HW3 ]
image: homework/hw3/hw3_nsoley1.png
featured: false
---

### What happens if I do or not not normalize and/or transform the gene expression data (e.g. log and/or scale) prior to dimensionality reduction?

I explored to see the differences in the scatter plot for both the dimensional reduction techniques and data normalization. Before normalizing for PCA, I observed that genes with larger expression values may dominate the principal components. Most of the variance can be seen only along PC1. Whereas normalization ensures that each gene contributes more equally to the principal components. Hence we can see different clusters of CBP1 expression spread across PC1 and PC2. Log transformation stabilizes variance, especially for highly expressed genes.

For t-SNE, without normalization higher expression are again dominating. t-SNE is known for preserving local structures, without normalization, it may be influenced by genes with large expression values. After normalization, I could see the clusters for both higher and lower expressed genes. Overall, my observation was that normalization in both the techniques stabilizes the variance and ensures that each gene contributes proportionally to the analysis.

### Write a description describing your data visualization using vocabulary terms from Lesson 1.

For all the plots, I used the geometric primitive of points to show the quantitative data for cells with CPB1 gene expression levels. I am visualizing quantitative data of the expression level of the CPB1 gene. The visual channel of color saturation is used to encode the expression levels of the CPB1 gene. I am also using transparency for depth in understanding, and size for an additional layer of information as visual channels. This allows the user to visualize the relationship between possible clusters of gene expression based on the transparency, intesity of the color, and size of the points.The transparency reveals overlapping points, emphasizing areas of higher density, while the color intensity provides insights into the intensity of gene expression.The reason behind this is that, the data after dimensionality reduction should be separated into clusters, the color hue encoding of the cells allows us to possibly see a trend in groups relating higher or lower gene expression.\
The gestalt principle of similarity is applied, where points with similar color are perceived as related groups. The gestalt principle of proximity is also in play since closer cells are perceived to be in a cluster. The side-by-side graphs effectively underscores the salient feature of normalization. The graph shows how normalization helps in stabilizing variance, ensuring equitable gene contribution, and unveiling potential trends in expression clusters.

### You must include the entire code you used to generate the figure so that it can be reproduced. You must provide attribution to external resources referenced (if any) in writing your code.

Used class notes to write the code:

```{r}
library(viridis)
library(patchwork)
library(Rtsne)

# data loading
data <- read.csv('/Users/nidhisoley/Desktop/GDataViz/eevee.csv.gz', row.names=1)
pos <- data[, 2:3]
gexp <- data[, 4:ncol(data)]

# top genes
top_gene <- names(sort(apply(gexp, 2, var), decreasing = TRUE)[1:1000]) 
gexp_filter <- gexp[, top_gene]
dim(gexp_filter)

# Applying PC to gexp without normalizing
pcs_without_norm <- prcomp(gexp_filter)

# Dataframe PC without norm and plot p1
df_pcs_without <- data.frame(pcs_without_norm$x)
gene <- gexp_filter$CPB1
p1 <- ggplot(df_pcs_without) +
  geom_point(aes(x = PC1, y = PC2, col = 0.8 * gene, alpha = 0.8 * gene, size = 0.8 * gene)) +
  scale_color_viridis(option = 'plasma') +
  guides(color = guide_legend(title = "CPB1 Expression"),
         alpha = guide_legend(title = "CPB1 Expression"),
         size = guide_legend(title = "CPB1 Expression")) +
  theme_classic()+  labs(title = "PCA without Normalization")


# Normalize the data and apply PCA and plot p2
gexp_norm <- log10(gexp_filter / rowSums(gexp_filter) * mean(rowSums(gexp_filter)) + 1)
pcs <- prcomp(gexp_norm)
df_pcs <- data.frame(pcs$x)
gene_2 <- gexp_norm$CPB1
p2 <- ggplot(df_pcs) +
  geom_point(aes(x = PC1, y = PC2, col = 0.8 * gene_2, alpha = 0.8 * gene_2, size = 0.8 * gene_2)) +
  scale_color_viridis(option = 'plasma') +
  labs(title = "PCA with Normalization") +
  guides(color = guide_legend(title = "CPB1 Expression\n(Log10 of Count)"),
         alpha = guide_legend(title = "CPB1 Expression\n(Log10 of Count)"),
         size = guide_legend(title = "CPB1 Expression\n(Log10 of Count)")) +
  theme_classic()

# t-SNE on pcs
emb1 <- Rtsne(pcs_without_norm$x[, 1:2], dim = 2, perplexity = 10)  # PCs without norm
emb2 <- Rtsne(pcs$x[, 1:2], dims = 2, perplexity = 10)  # PCs with norm
# tsne for PCs with norm
p3 <- ggplot(df_emb) +
  geom_point(aes(x = emb2.1, y = emb2.2, col = 0.8 * gene_2, alpha = 0.8 * gene_2, size = 0.8 * gene_2)) +
  scale_color_viridis(option = "turbo") +
  labs(title = "t-SNE with Normalization") +
  guides(color = guide_legend(title = "CPB1 Expression\n(Log10 of Count)"),
         alpha = guide_legend(title = "CPB1 Expression\n(Log10 of Count)"),
         size = guide_legend(title = "CPB1 Expression\n(Log10 of Count)")) +
  theme_classic()
# tsne for PCs without norm
p4 <- ggplot(df_emb) +
  geom_point(aes(x = emb1.1, y = emb1.2, col = 0.8*gene, alpha = 0.8* gene, size = 0.8*gene)) +
  scale_color_viridis(option = "turbo") +
  labs(title = "t-SNE without Normalization") +
  guides(color = guide_legend(title = "CPB1 Expression"),
         alpha = guide_legend(title = "CPB1 Expression"),
         size = guide_legend(title = "CPB1 Expression")) +
  theme_classic()

# Combine plots
final_plot <- p1 + p2 + p4 +  p3 + plot_annotation(tag_levels = 'a') + plot_layout(ncol = 2)
final_plot
```
