---
layout: post
title:  "PCA vs. tSNE"
author: Amanda Kwok
jhed: akwok1
categories: [ HW3 ]
image: homework/hw3/AmandaKwok.png
featured: false
---

### Whatâ€™s the difference if I perform linear or nonlinear dimensionality reduction to visualize my cells in 2D?
    The difference between linear and non-linear dimensionality reduction, specifically difference between PCA and tSNE, is that PCA linearly
projects the n-dimensional feature space down into 2D space that can be visualized and tSNE uses non linear cost methods to randomly group data points in a 2D space (as specified in the code). For PCA, this means that the 2D visualization will be on a new coordinate system (basis) generated by the 2 directions of variance, in my case PC1 and PC2 the largest directions of variance in the data. PCA is deterministic and unbiased in the sense that its algorithm is not focused on grouping objects but rather representing them in lower dimensional space based on deterministically calculated directions of variance. Thus, PCA results might not yield results that appear to cluster. On the other hand, tSNE has a goal of grouping objects with similar features by performing a kind of cost function trying to minimize the difference between two distributions as learned in class. As a result, the plots generated by tSNE will inherently resemble clusters as that as the main goal of the algorithm. From the plot generated in the code pasted below, you can see that tSNE does appear to cluster it's output more than PCA. 

### Write a description describing your data visualization using vocabulary terms from Lesson 1:
    For the plot on the left, labelled PC1 vs PC2, I used the geometric primitive of points to encode cells. That is, each point on the graph represents a cell that expresses a gene. The visualization is of quantitative data because the data is of gene expression counts/variance. In terms of visual channels, I encode the cell_area using color intensity. This allows the user to visualize the relationship between possible clusters of data and cell_area. The motivation behind this is that, say the data after dimensionality reduction is separated into clear groups, the color hue encoding of the cells allows us to possibly see a trend in groups relating to area. For instance, perhaps a certain cluster is significantly more intense in hue and perhaps this cluster also has high levels of gene expression which would allow us to draw a trend between area and gene expression counts. Another visual channel for this plot would be position. The axes of the plots are of principle components so the axes themselves don't hold obvious interpretations. However, cells that cluster near each other in terms of their position in the graph could be interpretted as part of a group or a class of cells. For the plot on the right, labelled tSNE, I also used to same geometric primitive of points to encode cells as well as the visual channels of color for cell area and position to show groups. Both plots have the same encodings as the visualizations are meant to be compared with each other in order to visualize the difference between linear and nonlinear dimensionality reduction techniques, respectively. In terms of gestalt principles, both feature the principle of similarity since the goal is to identify clusters, cells with similar positions and colors will be interpretted as part of the same class or group. It follows that the gestalt principle of proximity is also in play since closer cells are percieved as groups. In terms of the overall salience of these visualizations side by side, I think it makes salient the idea that nonlinear dimensionality reductions tend to form more obvious groupings than linear dimensionality reductions. The use of color channels isn't necessary to make this point salient however I included it to show that the color channel can be used to differentiate clusters and draw conclusions. 


### Code: Note: all the code I mainly got from in class work

# Hw 3

# install packages
install.packages('Rtsne')
install.packages('patchwork')
install.packages('ggplot2')

# access the libraries
library(Rtsne)
library(patchwork)
library(ggplot2)

# load the data
data <- read.csv('C:/Users/Amand/OneDrive/Documents/JHU-undergrad/Junior Year/Sem2/Genomic Data/Amanda-K/data/pikachu.csv.gz', row.names = 1)
dim(data)
data[1:5,1:5]

# process data
gexp <- data[,6:ncol(data)]
topgene <- names(sort(apply(gexp, 2, var), decreasing = TRUE)[1:15]) # apply to cols
gexpfilter <- gexp[,topgene]
dim(gexpfilter)

# Run PCA, find directions of max variance
pcs <- prcomp(gexpfilter)
plot(pcs$sdev[1:30])
dim(pcs$x)

# set the seed
set.seed(123)

# run tSNE
emb <- Rtsne(gexpfilter, check_duplicates = FALSE)
names(emb)
emb$Y

# generate plots
df <- data.frame(pcs$x, emb = emb$Y, gene = rowSums(gexpfilter), cell_area = data$cell_area)
p1 <- ggplot(df) + geom_point(aes(x = PC1, y = PC2, col = cell_area)) + 
  ggtitle('PC1 vs. PC2') + theme_classic() + theme(plot.title = element_text(hjust = 0.5))
p2 <- ggplot(df) + geom_point(aes(x = emb.1, y = emb.2, col = cell_area)) + 
  ggtitle('tSNE') + theme_classic() + theme(plot.title = element_text(hjust = 0.5))


p1 + p2 + plot_annotation(tag_levels = 'a') + plot_layout(ncol = 2) 

# theme(plot.title = element_text(hjust = 0.5)) line taken from: 
# https://stackoverflow.com/questions/40675778/center-plot-title-in-ggplot2
# the above line of code centers the titles